core/loop_mixin.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview The mixin that holds logic for setting a warning on a block if
+ * it is not inside a loop.
+ */
+'use strict';
+
+goog.module('Blockly.loopMixin');
+
+const Extensions = goog.require('Blockly.Extensions');
+
+
+/**
+ * This mixin adds a check to make sure the 'controls_flow_statements' block
+ * is contained in a loop. Otherwise a warning is added to the block.
+ * @mixin
+ * @augments Blockly.Block
+ * @readonly
+ */
+const CONTROL_FLOW_IN_LOOP_CHECK_MIXIN = {
+    /**
+     * List of block types that are loops and thus do not need warnings.
+     * To add a new loop type add this to your code:
+     * Blockly.LoopMixin.CONTROL_FLOW_IN_LOOP_CHECK_MIXIN.LOOP_TYPES.push('custom_loop');
```

Comment:
Update comment to refer to the new location, and add to renamings file as well.

Reply:
And add a comment telling them to talk to us if they want to use this :)

Reply:
Ok added a comment asking them to comment on [this issue](https://github.com/mit-cml/appinventor-sources/issues/2032) if they need to use this. And I updated `Blockly.LoopMixin` to be `Blockly.loopMixin`.